frontend/pkg/frontend/frontend.go

```diff
@@ -284,92 +284,102 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 	writer.WriteHeader(http.StatusOK)
 }
 
-func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request *http.Request) {
-	var err error
-
-	// This handles both PUT and PATCH requests. The only notable
-	// difference is PATCH requests will not create a new cluster.
-
+func (f *Frontend) ArmResourceCreate(writer http.ResponseWriter, request *http.Request) {
 	ctx := request.Context()
+	versionedInterface, doc, updating, err := f.getVersionedInterfaceAndDoc(ctx, writer, request)
+	if err != nil {
+		return
+	}
 
-	versionedInterface, err := VersionFromContext(ctx)
+	if updating {
+		f.logger.Info(fmt.Sprintf("document already exists for %s, use PATCH instead", doc.Key))
+		arm.WriteInternalServerError(writer)
+		return
+	}
```

Comment:
with regards to cluster docs, im not sure there is a scenario where we would actually update the doc to where it mattered if we were updating or not that we need to check it here?

The ArmResourceCreate call would only ever get routed to on PUT, in which if the cluster already exists i think the CS logic later on would handle errors trying to PUT a cluster that already exists and we can let that handle the error?

Thoughts @mbarnes @mjlshen ?

If we do want to explicitly check, we probably need to return the proper status code instead of a server error

Reply:
Updating with PUT is a valid ARM operation, even if it's not a valid Cluster Service operation.   We shouldn't be erroring out here.  If the resource exists and the PUT request passes static validation, pass it to the update handler.